src/frontend/screens/Settings/MapSettings/BGMapCard.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from "react";
+import { defineMessages, useIntl } from "react-intl";
+import { StyleSheet, View, Text } from "react-native";
+import { Bar } from "react-native-progress";
+
+import { MapServerStyleInfo } from "../../../sharedTypes";
+import { LIGHT_GREY, MAPEO_BLUE, MEDIUM_GREY } from "../../../lib/styles";
+import { useMapImportProgress } from "../../../hooks/useMapImportProgress";
+import { Pill } from "../../../sharedComponents/Pill";
+import { CUSTOM_MAP_ID, DEFAULT_MAP_ID } from "../../../hooks/useMapStyles";
+import MapThumbnail from "../../../sharedComponents/MapThumbnail";
+import { TouchableOpacity } from "../../../sharedComponents/Touchables";
+import { bytesToMegabytes } from ".";
+import { useNavigationFromRoot } from "../../../hooks/useNavigationWithTypes";
+
+const m = defineMessages({
+  currentMap: {
+    id: "sharedComponents.BGMapCard.currentMap",
+    defaultMessage: "Current Map",
+  },
+  abbrevMegabyte: {
+    id: "sharedComponents.BGMapCard.abbrevMegabyte",
+    defaultMessage: "MB",
+    description: "The abbreviation for megabyte",
+  },
+  unnamedStyle: {
+    id: "sharedComponents.BGMapCard.unamedStyle",
+    defaultMessage: "Unnamed Style",
+    description: "The name for the default map style",
+  },
+  waitingForImport: {
+    id: "sharedComponents.BGMapCard.waitingForImport",
+    defaultMessage: "Waiting for import…",
+    description:
+      "Progress bar message indicating that import is waiting to start",
+  },
+  importInProgress: {
+    id: "sharedComponents.BGMapCard.importInProgress",
+    defaultMessage: "Import in progress…",
+    description: "Progress bar message about the import being in progress",
+  },
+  errorOccurred: {
+    id: "sharedComponents.BGMapCard.errorOccurred",
+    defaultMessage: "Error occurred",
+    description: "Message describing that error occurred for map import",
+  },
+});
+
+const WithTopSeparation = ({ children }: React.PropsWithChildren<{}>) => (
+  <View style={{ marginTop: 10 }}>{children}</View>
+);
+
+interface BGMapCardProps extends MapServerStyleInfo {
+  isImporting: boolean;
+  isSelected: boolean;
+  onPress?: () => void;
+}
+
+export const BGMapCard = ({
+  isImporting,
+  isSelected,
+  name,
+  bytesStored,
+  id: styleId,
+  url: styleUrl,
+}: BGMapCardProps) => {
+  const { navigate } = useNavigationFromRoot();
+  const { formatMessage: t } = useIntl();
+
+  const importInfo = useMapImportProgress(styleId);
+
+  const showBytesStored =
+    styleId !== DEFAULT_MAP_ID && styleId !== CUSTOM_MAP_ID && !isImporting;
+  const showProgressBar = importInfo && importInfo.status !== "error";
+  const showProgressBarMessage =
```

Comment:
ah yeah the naming was more referring to the entity rather than the state e.g. ProgressBar as a noun rather than the type of bar 😅 happy to update if that's still unclear

Reply:
actually, this is only used in one spot, so maybe just easier to remove the variable declaration and do this inline 😄

Reply:
Great, thats would make it the clearest